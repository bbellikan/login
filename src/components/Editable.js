import React from 'react';
import PropTypes from 'prop-types';

export default class Editable extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            isEditing: this.props.isEditing || false
        };

        this.onblur_handler = this.onblur_handler.bind(this);
        this.onchange_handler = this.onchange_handler.bind(this);
    }

    onblur_handler() {
        this.setState({
            isEditing: !this.state.isEditing,
        });
    }

    onchange_handler() {

        let value = this.props.gridCol.inputType === 'checkbox' ? this.inputControl.checked : this.inputControl.value;
        this.props.updateCell(this.props.rowNumber, this.props.gridCol.header, value);
    }

    render() {
        let input = this.props.gridCol.inputType === 'checkbox'
            ? <label className='containerLabel'>
                <input id = {this.props.gridCol.number + '_' + this.props.rowNumber}
                       type='checkbox'
                       ref={(input) => {
                           this.inputControl = input;
                       }}
                       onChange={this.onchange_handler}
                       checked={this.props.value ? 'checked' : ''}
                />
                <span className='checkmark'
                      onBlur={this.onblur_handler}
                      autoFocus
                ></span>
            </label>
            : <input type={this.props.gridCol.inputType}
                     className='input'
                     ref={(input) => {
                         this.inputControl = input;
                     }}
                     value={this.props.value}
                     onChange={this.onchange_handler}
                     onBlur={this.onblur_handler}
                     autoFocus/>;

        if (this.state.isEditing) {

            return <div
                className='cell'
            >
                {input}
            </div>;
        }

        return <div
            className='cell'
            onClick={this.onblur_handler}
        >
            {this.props.gridCol.inputType === 'checkbox' ? input :this.props.value}
        </div>;
    }
}

Editable.propTypes = {
    gridCol:PropTypes.object.isRequired,
    rowNumber: PropTypes.number.isRequired,
    value:PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.string,
        PropTypes.number]),
    updateCell:PropTypes.func.isRequired
};
