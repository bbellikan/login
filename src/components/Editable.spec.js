import React from 'react';
import { shallow } from 'enzyme';
import Editable from './Editable';

const setup = (setupProps = {}) => {
    const defaultProps = {
        gridCol:{
            "number": 0,
            "order": 0,
            "header": "type",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        rowNumber: 0,
        value:'POS',
        updateCell:jest.fn(),
    };


    const props = {...defaultProps, ...setupProps};
    const wrapper = shallow(
        <Editable {...props}/>
    );

    return {
        props,
        wrapper
    };
};

describe('Editable', () => {
    it('renders without crashing', () => {
        const {wrapper} = setup();
        expect(wrapper).toMatchSnapshot();
    });

});
