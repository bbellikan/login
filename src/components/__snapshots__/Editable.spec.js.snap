// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editable renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editable
    gridCol={
        Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          }
    }
    rowNumber={0}
    updateCell={[Function]}
    value="POS"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "POS",
      "className": "cell",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "POS",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "POS",
        "className": "cell",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "POS",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
